model/src/main/proto/exploration.proto

```diff
@@ -322,27 +322,38 @@ message UserAnswer {
   WrittenTranslationContext written_translation_context = 6;
 }
 
-// Corresponds to a raw item selection answer that user has submitted.
+// Corresponds to a raw item selection answer that user has selected.
 message ItemSelectionRawAnswer {
   repeated int32 selected_indexes = 1;
 }
 
-// Corresponds to a last raw answer entered by user which is used to retain state on Config Change.
+// Corresponds to a raw representation of the current answer entered by the user which is used to
+// retain state on configuration changes.
 message RawUserAnswer {
   oneof answer_input_type {
-    // A raw answer entered by user in a text-based interaction.
+    // A raw answer entered by user in a text-based interactions.
     string textual_answer = 1;
-    // An indices selected in multiple choice selection.
-    int32 multiple_choice_selection_index = 2;
-    // A selected index in SelectionInteraction.
+    // A user's selection for item selection and multiple choice interactions.
     ItemSelectionRawAnswer item_selection = 3;
-    // A list of html content ids selected in DragAndDropInteraction.
-    ListOfSetsOfTranslatableHtmlContentIds drag_and_drop = 4;
-    // A selected image region in ImageRegionSelectionInteraction.
-    ClickOnImage image_region_selection = 5;
+    // A user's selection for drag and drop interactions.
+    DragAndDropRawAnswer drag_and_drop = 4;
+    // A user's selection for image region selection interaction.
+    ImageWithRegions.LabeledRegion image_region_selection = 5;
   }
 }
 
+// Corresponds to a current drag and drop answer
+message DragAndDropRawAnswer {
+  // list of group subtitled html
+  repeated GroupSubtitledHtml list_of_group_subtitled_html = 1;
+}
+
+// A list of subtitled html
+message GroupSubtitledHtml {
+  // list of subtitled html
+  repeated SubtitledHtml list_of_subtitled_htmls = 1;
```

Comment:
Why not use the existing structures for this? That seems cleaner (plus, I think you primarily need content IDs in these cases not the subtitled html objects).

Reply:
Yes, I have removed this extra protos. Thanks